engine/lite/components/request-response/src/main/scala/pl/touk/nussknacker/engine/lite/components/requestresponse/jsonschema/sources/JsonSchemaRequestResponseSourceFactory.scala

```diff
@@ -9,25 +9,30 @@ import pl.touk.nussknacker.engine.api.typed._
 import pl.touk.nussknacker.engine.api.typed.typing.Unknown
 import pl.touk.nussknacker.engine.api.{MetaData, NodeId}
 import pl.touk.nussknacker.engine.json.{JsonSchemaExtractor, SwaggerBasedJsonSchemaTypeDefinitionExtractor}
-import pl.touk.nussknacker.engine.requestresponse.api.openapi.RequestResponseOpenApiSettings.InputSchemaProperty
+import pl.touk.nussknacker.engine.requestresponse.api.openapi.RequestResponseOpenApiSettings.{InputSchemaProperty, OutputSchemaProperty}
 import pl.touk.nussknacker.engine.requestresponse.api.{RequestResponseSource, RequestResponseSourceFactory}
 
 class JsonSchemaRequestResponseSourceFactory extends RequestResponseSourceFactory with SingleInputGenericNodeTransformation[Source] {
 
-  override type State = Schema
+  override type State = (Schema, Schema)
 
   private val nodeIdDependency = TypedNodeDependency[NodeId]
   private val metaDataDependency = TypedNodeDependency[MetaData]
+
   override def nodeDependencies: List[NodeDependency] = List(nodeIdDependency, metaDataDependency)
 
   private val jsonSchemaExtractor = new JsonSchemaExtractor()
 
   override def contextTransformation(context: ValidationContext, dependencies: List[NodeDependencyValue])(implicit nodeId: NodeId): NodeTransformationDefinition = {
     case TransformationStep(Nil, _) =>
-      val determinedSchema = jsonSchemaExtractor.getSchemaFromProperty(InputSchemaProperty, dependencies)
-      val validationResult = determinedSchema.swap.toList.flatMap(_.toList)
-      val finalState = determinedSchema.toOption
-      val finalInitializer = determinedSchema.toOption.fold(new BasicContextInitializer(Unknown)) { schema =>
+      val inputSchema = jsonSchemaExtractor.getSchemaFromProperty(InputSchemaProperty, dependencies)
+      val outputSchema = jsonSchemaExtractor.getSchemaFromProperty(OutputSchemaProperty, dependencies)
+      val validationResult = inputSchema.swap.toList.flatMap(_.toList)
+      val finalState = List(inputSchema.toOption, outputSchema.toOption) match {
```

Comment:
Use ready method `inputSchema.product(outputSchema)` instead. Also `validationResult` doesn't have errors from outputSchema - after usage of product, you'll get it.